pyqtgraph/functions.py

```diff
@@ -1752,51 +1751,124 @@ def arrayToQPath(x, y, connect='all'):
             if first < len(x):
                 # Replace all non-finite entries from beginning of arr with the first finite one
                 idx[:first] = first
-                arr[1:-1] = arr[1:-1][idx]
+                arr[:] = arr[:][idx]
+
 
     # decide which points are connected by lines
     if eq(connect, 'all'):
-        arr[1:-1]['c'] = 1
+        polygon = arrayToQPolygonF(
+            x.astype(np.double),
```

Comment:
No need to convert to double if we relax the requirement in ```arrayToQPolygonF()```

Reply:
Just tested this with a variety of other numpy dtypes, and you're right, it still works....

Reply:
I'm assuming I still need to keep this guy tho `memory = np.frombuffer(buffer, np.double).reshape((-1, 2))`

Reply:
Yes, that is the destination buffer, which is defined to be doubles.